tasks/retrieval_deal.go

```diff
@@ -174,3 +198,43 @@ func (rp *_RetrievalTask__Prototype) Of(minerParam string, payloadCid string, ca
 	}
 	return &rt
 }
+
+// ParseStageTimeouts parses "StageName=timeout" strings into a map of stage
+// name to timeout duration.
+func ParseStageTimeouts(timeoutSpecs []string) (map[string]time.Duration, error) {
+	var stageTimeouts map[string]time.Duration
+
+	// Parse all stage timeout durations
+	timeouts := map[string]time.Duration{}
+	for _, spec := range timeoutSpecs {
+		parts := strings.SplitN(spec, "=", 2)
+		if len(parts) != 2 {
+			return nil, fmt.Errorf("invalid stage timeout specification: %s", spec)
+		}
+		stage := parts[0]
+		timeout := parts[1]
+		d, err := time.ParseDuration(strings.TrimSpace(timeout))
+		if err != nil || d < time.Second {
+			return nil, fmt.Errorf("invalid value for stage %q timeout: %s", stage, timeout)
+		}
+		timeouts[strings.ToLower(strings.TrimSpace(stage))] = d
```

Comment:
you probably want to error if there are duplicates

Reply:
Done